### DISCUSSION TITLE
Create a new decorator

### DISCUSSION THREAD

**Discussion by survuvi** [modular/modular#216] [Q&A] [answered]
What's not working?

```python

# create my hello decorator
fn my_hello_decorator(func):
    fn wrapper():
        print("Hello,")
        func()
        print("Bye Bye!")
    return wrapper

# use my hello decorator
@my_hello_decorator
fn print_name():
    print("my name is Mark")

print_name()

```

The output should be:
Hello,
my name is Mark
Bye Bye!

Debug:
_error: Expression [5]:5:23: 'fn' parameter type must be specified
fn my_hello_decorator(func):

error: Expression [5]:10:12: 'fn() capturing -> None' value cannot be converted to 'None' in return value
    return wrapper_

**Comment by daydun** [comment, marked as answer]
The error message is solved by giving the function correct types:
```py
fn my_hello_decorator(func: fn() -> None) -> fn() capturing -> None:
    fn wrapper():
        print("Hello,")
        func()
        print("Bye Bye!")
    return wrapper

@my_hello_decorator
fn print_name():
    print("my name is Mark")

print_name()
```
mojo does not yet support custom decorators though, so it will just print
```
my name is Mark
```

**Reply by survuvi** [reply]
Thank you

**Comment by iamtimdavis** [comment]
Thanks @DayDun for this answer! 🔥
